text/0000-more-env-info.md

```diff
@@ -0,0 +1,46 @@
+- Feature Name: more-env-info
+- Start Date: 2017-08-09
+- RFC PR:
+- Rust Issue:
+
+# Summary
+[summary]: #summary
+
+Introduce more potentially useful environment-related functions to `std::env`.
+
+# Motivation
+[motivation]: #motivation
+
+Programs sometimes need to know about their working environment to do their job properly. For example, linebreak convention differs for Windows and \*NIX. Such discrepancy can lead to problems easily, especially when a program needs to communicate with aged third-party libraries.
+
+As a system programming language, it would be good for Rust to know the system it works on.
+
+# Detailed design
+[design]: #detailed-design
+
+This RFC would like to introduce the following fields to `std::env::consts`:
+
+```rust
+/// Conventional linebreak of current platform.
+pub const LINEBREAK: &'static str;
+/// Word size in bits the program has been compiled into.
+/// Commonly 32 or 64.
+pub const WORD_SIZE: &'static u32;
```

Comment:
This should be a `u32` and not a `&u32`.

Also I think it's much more straightforward to use `std::mem::size_of<usize>()` (which gives you the bytes, but that's usually what you want anyway I think, also a `*8` doesn't warrant a new constant in the stdlib imo.